lib/filesystem.js

```diff
@@ -101,12 +101,12 @@ class Filesystem {
   insertLink (p) {
     const symlink = fs.readlinkSync(p)
     const parentPath = path.dirname(p)
-    const link = path.relative(fs.realpathSync(this.src), path.join(parentPath, symlink))
+    const link = path.resolve(parentPath, symlink)
```

Comment:
The modification here will cause the following if statement to always evaluate to false.

Reply:
Another way to write it is like this, but it doesn't pass the test. I don't know much about the working logic of asar, I only know that it can be packaged on mac.
```
  insertLink (p) {
    const symlink = fs.readlinkSync(p)
    const parentPath = path.dirname(p)
    const link = path.relative(fs.realpathSync(this.src), path.relative(path.join(parentPath, symlink)))
    if (link.substr(0, 2) === '..') {
      throw new Error(`${p}: file "${link}" links out of the package`)
    }
    const node = this.searchNodeFromPath(p)
    node.link = link
    return link
  }
```

Reply:
I mean using `realpathSync` 😄. e.g.

```js
const link = path.relative(fs.realpathSync(this.src), path.realpathSync(path.join(parentPath, symlink)))
```

Reply:
Yes, I wrote it wrong, but that doesn't pass the test.

Reply:
Thank you for your reply! I may need some time to investigate this issue. Currently, this PR cannot be merged as it would invalidate the following code: https://github.com/electron/asar/blob/d50b03f48ce60383a62cf322d7868bc2a183a41e/lib/filesystem.js#L105-L107

Reply:
Can you provide a reproducible GitHub repo for this? Like: https://github.com/electron/asar/pull/301#issuecomment-1985253352

Reply:
I understand this.
This is the error I encountered while following the [instructions](
https://www.electronforge.io/core-concepts/build-lifecycle) using [Electron Forge](https://www.electronforge.io/) to build.
I use a macbook, maybe other systems may have no problem.